### PR TITLE
change dir name, adds hexagonal architecture

### PR DESCRIPTION
**Pull request by crocme10**

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  concerned with translating queries from the user to Elasticsearch, and
+  Elasticsearch's response back to the user.
+
+* **Binaries** for each file format that we ingest in the backend through Mimir.
+  These provide a command line interface which make use of *Places* and *Mimir*
+  to validate and enrich the data prior to insertion into the backend.
+
+The context diagram showing these two components is seen below:
+
+![Context Diagram](/docs/img/architecture/context-diagram.svg)
```
Comment (amatissart) [inline docs/architecture.md:31]: This image is not present on this branch.

File: docs/architecture.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+categorized as:
+1. **primary** (or driving) when they make a request to the domain,
+2. **secondary** (or driven) when the domain makes a request to them.
+
+For example, a user interacts with the domain through a primary port, where as
+the domain will persist data to a database through a seconday port. It is
+customary to represent primary port on the left, and secondary port on the
+right, as seen in the picture below.
+
+![Hexagonal Architecture 1](/docs/img/architecture/hexagon-generic-1.2.svg)
```
Comment (amatissart) [inline docs/architecture.md:54]: The same here

File: docs/architecture.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+why they are also called **controllers**. They are injected in their
+constructor an object (which we'll call a **service**) whose class implements
+the primary port's interface that the adapter requires. Sometimes primary
+adapters are also called controllers.
+
+Secondary adapters, on the other hand, directly implement a secondary port's interface.
+
+The following diagram shows these components interacting together:
+
+![Hexagonal Architecture 2](/docs/img/architecture/hexagon-generic-2.2.svg)
```
Comment (amatissart) [inline docs/architecture.md:73]: and here.

**Review by amatissart — COMMENTED** [review]
Nice! That's a great overview of how the different components work together.

Some images linked from "architecture.md" are missing though.

File: docs/architecture.md (hunk 3 above)
Comment (crocme10) [reply docs/architecture.md]: fixed

**Pull request merged**